app/views/partials/_post_table.html.erb

```diff
@@ -1,79 +1,110 @@
+
+
 <div class="posts-table index-table">
-	<% unless post_details.empty? %>
+	<% unless posts.empty? %>
 		<table>
 			<thead>
 				<tr>
 					<td>Posted By</td>
-					<td colspan="2">Address</td>
-					<!-- <td>Needs</td> -->
+					<td colspan="1">Address</td>
 					<td>Post Needs</td>
 					<td>Status</td>
 					<td>Actions</td>
 				</tr>
 			</thead>
 			<tbody>
-				<% post_details.each do |post| %>
-					<% post_needs = Post.find(post['id']).post_needs %>
-					<tr>
-						<td>
-							<p class="mobile-only table-heading">Posted By</p>
-							<%= link_to post["poster_name"], user_path(id: post["poster_id"]) %>
-							<small><%= format_time(post["created_at"].to_time) %></small>
-						</td>
-						<td colspan="2">
-							<p class="mobile-only table-heading">Address</p>
-							<%= post["street_1"] %><br/>
-							<% unless post["street_2"].empty? %>
-								<%= post["street_2"] %><br/>
-							<% end %>
-							<%= post["city"] %>, <%= post["state"] %> <%= post["zip"] %>
-						</td>
-						<td>
-							<p class="mobile-only table-heading">Post Needs</p>
-							<ul>
-								<% post_needs.each do |pn| %>
-									<li>
-										<%= pn.need.name %>
-										<% if pn.complete? %>
-											✓
-										<% end %>
-									</li>
+				<% posts.each do |p| %>
+				    <%unless p.nil?%>
+				    <!-- for now do not show the cancelled requests -->
+						<% post_needs = p.post_needs %>
+						<tr>
+							<td>
+								<p class="mobile-only table-heading">Posted By</p>
+								<%= link_to "#{p.poster.first_name} #{p.poster.last_name}", user_path(p.poster) %></br>
```

Comment:
The reason why I wrote the hacky get_post_details method is because that way, the user's information doesn't get transferred past the model layer. What you're doing here is making database calls from the view layer for user data-- we probably don't want to do that. In other words, I'm restricting what data the browser gets in this method for data security purposes. I understand you probably thought this was hacky and wanted to reimplement this, but let's not expose the user's info to the DOM.

Reply:
So do I need to change it? I am using the pagination and I don't think that pagination can be used on arrays...
If more requests are posted, will it be very inconvenient for the users if pagination is not used?